upgrade/steps.go

```diff
@@ -845,8 +845,8 @@ func migrationSteps(ctx Context, repo ProviderRepo, providerName string, descrip
 	return steps, nil
 }
 
-func AddAutoAliasing(ctx Context, repo ProviderRepo, providerName string) (step.Step, error) {
-	providerName = strings.TrimPrefix(providerName, "pulumi-")
+func AddAutoAliasing(ctx Context, repo ProviderRepo) (step.Step, error) {
+	providerName := strings.TrimPrefix(repo.name, "pulumi-")
```

Comment:
I still don't understand. Provider name is a user option flag. We shouldn't need to trim `pulumi-` to get it. If this is us applying the default, let's apply it earlier and so consuming it here is identical to consuming the user option.

Reply:
I believe the user input is {org}/{repo}, so say `repo.org == `pulumi`, `repo.name` == `pulumi-xyz`. Here, we assume the subdirectory should be `pulumi-resource-xyz` containing the metadata file ([following the boilerplate](https://github.com/pulumi/pulumi-tf-provider-boilerplate/tree/main/provider/cmd), so we trim `pulumi-` from `repo.name`. 
Before, providerName was just the repo name so I think this should be equivalent?

Reply:
Tested on https://github.com/pulumi/pulumi-pagerduty/pull/194

Reply:
What do we do if `repo.name` is `my-great-provider`?

If this is invalid, lets make sure that we assert that early in the process.

Reply:
Good point on the assertion. Looking at our [pulumiverse](https://github.com/orgs/pulumiverse/repositories?type=all) providers, i think it's a fair assumption (requirement?) that provider names should start with `pulumi-`.